### PR TITLE
5.x CHANGELOG + Migration guide + version update

### PR DESCRIPTION
**Pull request by bpoplauschi**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

### Pull Request Description

Bumped the `5.x` branch version to `5.0.0-beta` -  this will be the version we release first.
Updating `CHANGELOG.md` + adding `SDWebImage-5.0-Migration-guide.md`

### CODE REVIEW

File: Docs/SDWebImage-5.0-Migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+As a major release, following [Semantic Versioning](http://semver.org/) conventions, 5.0 introduces several API-breaking changes with its new architecture.
+
+.... RELEASE_SUMMARY ....
+
+This guide is provided in order to ease the transition of existing applications using SDWebImage 4.X to the latest APIs, as well as explain the design and structure of new and changed functionality.
+
+### Requirements: iOS 8, Mac OS X 10.10, watchOS 2, tvOS 9, Xcode 7.3
+
+SDWebImage 5.0 officially supports iOS 8 and later, Mac OS X 10.10 and later, watchOS 2 and later and tvOS 9 and later.
+It needs Xcode 7.3 or later to be able to build everything properly.
```
Comment (dreampiggy) [inline Docs/SDWebImage-5.0-Migration-guide.md]: ...Actually, we already upgrate to use Xcode 8, or many feature like APNG, class property (For Swift user API) will fail. Please update this :)

Comment (bpoplauschi) [reply Docs/SDWebImage-5.0-Migration-guide.md]: Done. I copy-pasted the Xcode 7.3 value until I could determine if an update is required. c38b731

**Comment by bpoplauschi** [comment]
@dreampiggy please properly review the changelog and migration guide. I must say I can't follow all the changes made in this big release, so I'm not sure I covered them properly. Please take proper time to do this, it's very important for our users.

**Comment by dreampiggy** [comment]
@bpoplauschi Can we share the migration guide firstlly on the wiki page. To allow quicklly modification and check ? After all finished, we can copy that and paste on the Git repo.

Because each time you edit and push a new commit will cause CI to build...

**Comment by bpoplauschi** [comment]
@dreampiggy I like this idea. Let's keep working on the wiki until we are done :)

**Comment by dreampiggy** [comment]
@bpoplauschi I use [objc-diff](http://codeworkshop.net/objc-diff/) to generate the API-diff from current `master` with `5.x`. It looks great and more clear. I upload it to you (And you can also download that command line to try yourself.)

Since we will move all FLAnimatedImage related code to another repo. This one I'll check at the last with another PR to remove it.

[api-diff.zip](https://github.com/rs/SDWebImage/files/2188925/api-diff.zip)

**Comment by bpoplauschi** [comment]
@dreampiggy this tool is so nice, had no idea something like that exists. It will help a lot. I was doing the diff manually

**Comment by dreampiggy** [comment]
@bpoplauschi I update the migration guide on wiki page. With some changes for API-breaking. And also a little introduce for the 5.0 brand new feature. The last chapter is the full api diff link. We can place that `apidiff.html && apidiff.css`, into the git repo folder `SDWebImage/Docs/Diff/5.0/apidiff.html`, to allow user check that instead if they really want to know the detail changes.

**Comment by bpoplauschi** [comment]
I think we can just remove my list and use the generated one (as you said, in the repo), it's better organised and we can always regenerate it. What do you think?

**Comment by dreampiggy** [comment]
@bpoplauschi I guess maybe we just need to list the API breaking (Which need change code) changes in that Migration guide. And provide the full api diff as well in the repo.

Why to do, because our hand-writed change also provide a `replacement advice`. For example, when user see this change:

#### SDImageCache
- `shouldDecompressImages` removed. Use `SDImageCacheAvoidDecodeImage` in cache options instead

They can immediately know how to change their code to migration this. Because some API change replacement may not so obvious from the full api diff page. (Another example, is `SDWebImageManagerDelegate`, which was removed because we use a better `SDImageTransformer` API instead). 😅

**Comment by bpoplauschi** [comment]
@dreampiggy very good point. I will do it.

**Comment by bpoplauschi** [comment]
Ok, I brought the changes from the wiki to this PR. Looks much better.

**Comment by dreampiggy** [comment]
@bpoplauschi Migration guide Wiki updated. Could you please update the latest one ? (I'm not a native English seaker, which may contains other grammar mistake, could you please also check and fix that ?) 😅

**Comment by bpoplauschi** [comment]
@dreampiggy I've added the latest changes from Wiki + small changes to make things for English friendly :)
Please take a look

**Comment by bpoplauschi** [comment]
@dreampiggy any blocker for merging this one?

**Comment by dreampiggy** [comment]
@bpoplauschi No anything block. I approve this. Maybe we can finally update the wiki and README.md (for example, something in the CHANGELog.md). But anything, no extra big modification is need since it's just a beta release

**Comment by bpoplauschi** [comment]
Ok, time to merge this one.

**Pull request merged**